operator/pkg/controllers/resourcesyncer/deployment.go

```diff
@@ -354,15 +356,22 @@ func (d *deploySyncer) sidecarInitContainer(deploy *v1.Deployment) (corev1.Conta
 	if err != nil {
 		return initContainer, err
 	}
-
 	appContainer, err := d.getAppContainer(deploy)
 	if err != nil {
 		return initContainer, err
 	}
 
-	if len(appContainer.Ports) != 0 {
+	applicationPort, ok := d.meshDeployment.Spec.Service.Labels[sideCarApplicationPortLabel]
```

Comment:
use `len(labels[applicationPort] == 0` to judge.

Reply:
Done